### PR TITLE
xds: use load assignment endpoint address in Cluster as the DNS hostname for LOGICAL_DNS

### PR DESCRIPTION
**Pull request by voidzcy**
This change includes: 

- parse the _single_ endpoint address from the embedded `Cluster` resource in CDS responses as the DNS hostname for LOGICAL_DNS cluster and include it in CdsUpdate being notified to the CDS LB policy.
- propagate the DNS hostname to the cluster_resolver LB policy via its LB config (DiscoveryMechanism for LOGICAL_DNS cluster).
- cluster_resolver LB policy takes the DNS hostname from the DiscoveryMechanism for LOGICAL_DNS cluster and use it as the name for DNS resolution.

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java [hunk 1]
```diff
@@ -560,7 +558,7 @@ public void run() {
                 addresses.add(eag);
               }
               PriorityChildConfig priorityChildConfig = generateDnsBasedPriorityChildConfig(
-                  name, edsServiceName, lrsServerName, maxConcurrentRequests, tlsContext,
+                  name, null, lrsServerName, maxConcurrentRequests, tlsContext,
```
Comment (dapengzhang0) [inline xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java:561]: The second argument of `generateDnsBasedPriorityChildConfig()` can be removed.

**Comment by voidzcy** [comment]
This can only be merged after TD's change is rolled out. To avoid introducing more conflicts, would update this after #8176 is merged.

File: xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java (hunk 1 above)
Comment (voidzcy) [reply xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java:565]: Good catch! Eliminated it.

**Comment by voidzcy** [comment]
Conflicts resolved. PTAL.

**Review by dapengzhang0 — APPROVED** [review]

**Pull request merged**